### PR TITLE
fix update leader

### PR DESCRIPTION
**Pull request by zhexuany**
After this, logic of updating region leader should be correct.

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tikv-client-lib-java/105)

### CODE REVIEW

File: src/main/java/com/pingcap/tikv/region/RegionManager.java [hunk 1]
```diff
@@ -192,10 +201,12 @@ public void onRequestFail(long regionID, long storeID) {
...
             }
           } catch (InterruptedException | ExecutionException e) {
-            e.printStackTrace();
+            invalidateRegion(regionID);
           }
           // store's meta may be out of date.
           invalidateStore(storeID);
         });
+    // missing one last step here, need remove these region cache's store id is store id
+    // in go code, remove region cache share same store id.
```
Comment (zhexuany) [inline src/main/java/com/pingcap/tikv/region/RegionManager.java]: shall we just remove all cache? or it does not matter since load region is lazy.

Comment (ilovesoup) [reply src/main/java/com/pingcap/tikv/region/RegionManager.java]: lazy is for first time and to perform as async. revisit will be cached. no cache will cause too much traffic for pd. But it seems lazy can be change to eager. But change the exception handle back to loading cache's lambda.

Comment (zhexuany) [reply src/main/java/com/pingcap/tikv/region/RegionManager.java]: Yeah. that is my concern too. I could not find a way to invalidate element associated with same store id from `LoadCache`.

Comment (ilovesoup) [reply src/main/java/com/pingcap/tikv/region/RegionManager.java]: according to loadingcache's semantics, we might change it to eager loading, and if exception happens, just don't change cache status.

Comment (zhexuany) [reply src/main/java/com/pingcap/tikv/region/RegionManager.java]: LoadCache doest not fit the semantics in this particular case.  We may need write our own cache. That is rather quite lot work. I guess the temporary solution we have is invalidate all cache. At least, this way will not give us any error.

**Review by ilovesoup — APPROVED** [review]
LGTM for now. I will merge and test on another branch with some other change together.

**Comment by ilovesoup** [comment]
Also deleted channel close since underlying channel would be kept alive unless non-idle by GRPC itself (idle timeout added).
Recent tests on single machine vs multiple kv shows the concurrent channel racing issue when close channel due to problem discussed here 
https://groups.google.com/forum/#!topic/grpc-io/6e0GFRSWit8
 It seems channel shutdown is not graceful as said in grpc code comment.
So we simply remove it.

File: src/main/java/com/pingcap/tikv/region/RegionManager.java [hunk 2]
```diff
@@ -34,168 +32,218 @@
...
-    try {
-      TiRegion region = regionCache.getUnchecked(regionId).get();
-      keyToRegionIdCache.remove(makeRange(region.getStartKey(), region.getEndKey()));
-    } catch (Exception ignore) {
-    } finally {
-      lock.writeLock().unlock();
-      regionCache.invalidate(regionId);
+    @SuppressWarnings("unchecked")
+    /**
+     * Remotes region associated with regionId from regionCache.
```
Comment (zhexuany) [inline src/main/java/com/pingcap/tikv/region/RegionManager.java]: should be “removes”？

**Pull request merged**